File: 1528-KidsWithTheGreatestNumberOfCandies/1528-KidsWithTheGreatestNumberOfCandies.py
# Last updated: 6/17/2025, 10:25:45 PM
class Solution:
    def kidsWithCandies(self, candies, extraCandies):
        max_candy = max(candies)  
        result = []  
        
        for candy in candies:
            if candy + extraCandies >= max_candy:  
                result.append(True)
            else:
                result.append(False)
        
        return result

File: 1528-KidsWithTheGreatestNumberOfCandies/1528-KidsWithTheGreatestNumberOfCandies.java
// Last updated: 8/23/2025, 6:49:50 PM
class Solution {
    public List<Boolean> kidsWithCandies(int[] candies, int extraCandies) {
        List<Boolean> result = new ArrayList<>();
        int maximum = 0;
        for(int i = 0; i < candies.length; i++){
            if(candies[i] > maximum){
                maximum = candies[i];
            }
        }
        for(int i = 0; i < candies.length; i++){
            if(candies[i] + extraCandies >= maximum){
                result.add(true);
            }
            else{
                result.add(false);
            }
        }
        return result;
    }
}